src/Illuminate/Support/Traits/EnumeratesValues.php

```diff
@@ -492,6 +492,28 @@ public function where($key, $operator = null, $value = null)
         return $this->filter($this->operatorForWhere(...func_get_args()));
     }
 
+    /**
+     * Filter items where the given key is not null.
+     *
+     * @param  string  $key
+     * @return static
+     */
+    public function whereNull($key)
+    {
+        return $this->where($key, null);
+    }
+
+    /**
+     * Filter items where the given key is null.
+     *
+     * @param  string  $key
+     * @return static
+     */
+    public function whereNotNull($key)
+    {
+        return $this->where($key, '!=', null);
```

Comment:
These methods would have to use a strict comparison operators `'==='` and `'!==`'.

These PHP comparisons are true: `'' == null`, `0 == null`, `0.0 == null`.

```php
Illuminate\Support\Collection::macro('whereNull', function ($key) {
  return $this->where($key, null);
});

Illuminate\Support\Collection::macro('whereNotNull', function ($key) {
    return $this->where($key, '!=', null);
});

$items = [
    ['falsy' => ''],
    ['falsy' => 0],
    ['falsy' => 0.0],
];

collect($items)->whereNull('falsy')->all();
// [
//   [
//     "falsy" => "",
//   ],
//   [
//     "falsy" => 0,
//   ],
//   [
//     "falsy" => 0.0,
//   ],
// ]

collect($items)->whereNotNull('falsy')->all();
// []
```

Reply:
Good catch!